account_invoice_margin/report/account_invoice_report.py

```diff
@@ -1,14 +1,45 @@
 # © 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class AccountInvoiceReport(models.Model):
     _inherit = "account.invoice.report"
 
     margin = fields.Float(string="Margin", readonly=True)
+    margin_percent = fields.Float(string="Margin (%)", readonly=True)
 
     def _select(self):
         select_str = super()._select()
-        return "%s, SUM(line.margin_signed) AS margin" % select_str
+        return (
+            "%s, SUM(line.margin_signed) AS margin, "
+            "AVG(move.margin_percent)  AS margin_percent" % select_str
+        )
+
+    @api.model
+    def read_group(
+        self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True
+    ):
+        res = super(AccountInvoiceReport, self).read_group(
+            domain,
+            fields,
+            groupby,
+            offset=offset,
+            limit=limit,
+            orderby=orderby,
+            lazy=lazy,
+        )
+        if "margin" not in fields:
+            return res
+
+        full_fields = all(x in fields for x in {"margin", "margin_percent"})
```

Comment:
Shouldn't this be?

```suggestion
        full_fields = all(x in fields for x in {"price_total", "margin_percent"})
```

Reply:
current version of fix working well as per this example: https://github.com/OCA/margin-analysis/pull/102#discussion_r688725059 or is there any additional example to verify?

I will do the same testing once runbot is up.

Reply:
The logic is that we should have here the measures that are involved in the computation, isn't it?

Reply:
Yes, that's true. IMO both approach will give us same output. I am open to change field from `margin` to `price_total`

Reply:
Let's do it for this version then.

Reply:
I have added as a separate commit.